includes/import/abstract-wc-product-importer.php

```diff
@@ -209,91 +170,107 @@ protected function get_product_object( $data ) {
 	 */
-	protected function save_product_data( $product, $data ) {
+	protected function process_item( $data ) {
+		try {
+			$object   = $this->get_product_object( $data );
+			$updating = false;
 
-		// Name.
-		if ( isset( $data['name'] ) ) {
-			$product->set_name( wp_filter_post_kses( $data['name'] ) );
-		}
+			if ( is_wp_error( $object ) ) {
+				return $object;
+			}
 
-		// Description.
-		if ( isset( $data['description'] ) ) {
-			$product->set_description( wp_filter_post_kses( $data['description'] ) );
-		}
+			if ( $object->get_id() && 'importing' !== $object->get_status() ) {
+				$updating = true;
+			}
 
-		// Short description.
-		if ( isset( $data['short_description'] ) ) {
-			$product->set_short_description( wp_filter_post_kses( $data['short_description'] ) );
-		}
+			$result = $object->set_props( array_diff_key( $data, array_flip( array( 'meta_data', 'raw_image_id', 'raw_gallery_image_ids', 'raw_attributes' ) ) ) );
 
-		// Status.
-		if ( isset( $data['published'] ) ) {
-			$product->set_status( $data['published'] ? 'publish' : 'draft' );
-		} elseif ( 'importing' === $product->get_status() ) {
-			$product->set_status( 'publish' );
-		}
+			if ( is_wp_error( $result ) ) {
+				throw new Exception( $result->get_error_message() );
+			}
 
-		// Slug.
-		if ( isset( $data['slug'] ) ) {
-			$product->set_slug( $data['slug'] );
-		}
+			if ( 'variation' === $object->get_type() ) {
+				$this->set_variation_data( $object, $data );
+			} else {
+				$this->set_product_data( $object, $data );
+			}
 
-		// Comment status.
-		if ( isset( $data['reviews_allowed'] ) ) {
-			$product->set_reviews_allowed( $data['reviews_allowed'] );
-		}
+			$this->set_image_data( $object, $data );
+			$this->set_meta_data( $object, $data );
 
-		// Virtual.
-		if ( isset( $data['virtual'] ) ) {
-			$product->set_virtual( $data['virtual'] );
-		}
+			if ( 'importing' === $object->get_status() ) {
+				$object->set_status( 'publish' );
+			}
 
-		// Tax status.
-		if ( isset( $data['tax_status'] ) ) {
-			$product->set_tax_status( $data['tax_status'] );
-		}
+			$object = apply_filters( 'woocommerce_product_import_pre_insert_product_object', $object, $data );
+			$object->save();
 
-		// Tax Class.
-		if ( isset( $data['tax_class'] ) ) {
-			$product->set_tax_class( $data['tax_class'] );
+			return array(
+				'id'      => $object->get_id(),
+				'updated' => $updating,
+			);
+		} catch ( Exception $e ) {
+			return new WP_Error( 'woocommerce_product_importer_error', $e->getMessage(), array( 'status' => $e->getCode() ) );
 		}
+	}
 
-		// Catalog Visibility.
-		if ( isset( $data['catalog_visibility'] ) ) {
-			$product->set_catalog_visibility( $data['catalog_visibility'] );
+	/**
+	 * Convert raw image URLs to IDs and set.
+	 *
+	 * @param WC_Product $product
+	 * @param array $data
+	 */
+	protected function set_image_data( &$product, $data ) {
+		// Image URLs need converting to IDs before inserting.
+		if ( isset( $data['image_id'] ) ) {
```

Comment:
@mikejolley should this be `$data['raw_image_id']` and `$data['raw_gallery_image_ids']`? I wasn't able to find anything in here that converts from `$data['raw_image_id'] => $data['image_id']`.